contracts/vaults/RibbonThetaVault.sol

```diff
@@ -286,89 +268,66 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
     }
 
     /**
      * @notice Mints the vault shares to the msg.sender
      * @param amount is the amount of `asset` deposited
      */
     function _deposit(uint256 amount) private {
-        uint256 totalWithDepositedAmount = totalBalance();
+        uint16 _round = round;
+        uint256 totalWithDepositedAmount = totalBalance().add(amount);
+
+        require(amount < MAX_UINT128, "Overflow");
         require(totalWithDepositedAmount < cap, "Exceed cap");
         require(
             totalWithDepositedAmount >= minimumSupply,
             "Insufficient balance"
         );
 
-        // amount needs to be subtracted from totalBalance because it has already been
-        // added to it from either IWETH.deposit and IERC20.safeTransferFrom
-        uint256 total = totalWithDepositedAmount.sub(amount);
-
-        uint256 shareSupply = totalSupply();
-
-        // Following the pool share calculation from Alpha Homora:
-        // solhint-disable-next-line
-        // https://github.com/AlphaFinanceLab/alphahomora/blob/340653c8ac1e9b4f23d5b81e61307bf7d02a26e8/contracts/5/Bank.sol#L104
-        uint256 share =
-            shareSupply == 0 ? amount : amount.mul(shareSupply).div(total);
-
-        require(shareSupply.add(share) >= minimumSupply, "Insufficient supply");
-
-        emit Deposit(msg.sender, amount, share);
-
-        _mint(msg.sender, share);
-    }
-
-    /**
-     * @notice Withdraws ETH from vault using vault shares
-     * @param share is the number of vault shares to be burned
-     */
-    function withdrawETH(uint256 share) external nonReentrant {
-        require(asset == WETH, "!WETH");
-        uint256 withdrawAmount = _withdraw(share);
-
-        IWETH(WETH).withdraw(withdrawAmount);
-        (bool success, ) = msg.sender.call{value: withdrawAmount}("");
-        require(success, "Transfer failed");
-    }
+        emit Deposit(msg.sender, amount, _round);
 
-    /**
-     * @notice Withdraws WETH from vault using vault shares
-     * @param share is the number of vault shares to be burned
-     */
-    function withdraw(uint256 share) external nonReentrant {
-        uint256 withdrawAmount = _withdraw(share);
-        IERC20(asset).safeTransfer(msg.sender, withdrawAmount);
-    }
+        VaultDeposit.DepositReceipt storage depositReceipt =
+            depositReceipts[msg.sender];
 
-    /**
-     * @notice Burns vault shares and checks if eligible for withdrawal
-     * @param share is the number of vault shares to be burned
-     */
-    function _withdraw(uint256 share) private returns (uint256) {
-        (uint256 amountAfterFee, uint256 feeAmount) =
-            withdrawAmountWithShares(share);
+        // If we have a pending deposit in the current round, we add on to the pending deposit
+        if (_round == depositReceipt.round) {
+            // No deposits allowed until the next round
+            require(!depositReceipt.processed, "Processed");
```

Comment:
so receipts will **only** be processed after weeks shorts are closed?

Reply:
They can only be processed in the new round. There are a few ways to get them processed:
- New deposit in the next round
- User calls `redeemDeposit` manually to redeem their shares